WordPress/gutenberg
# Remove anchor support from dynamic blocks

Fixes: #48232
Fixes: #49382
Fixes: #49578

Related to: #48438

## What?
This PR reverts anchor support for dynamic blocks added by #44771.

## Why?
In #44771, anchor support was added to dynamic blocks, but static blocks with existing anchor were broken (See #48232, #49382, and #49578). This feature was originally planned to be included in WordPress 6.2, but #48232 was the first to report this issue during the Beta phase. Therefore, by #48592, this feature was excluded in the WordPress 6.2 release.

However, this feature still remains on the Gutenberg plugin, which will be bundled with WordPress 6.3 and will cause similar problems. A solution was attempted in #48438, but [we decided to revert anchor support for dynamic blocks in the Gutenberg plugin](https://github.com/WordPress/gutenberg/pull/48438#issuecomment-1486234187) as well, in order to find a more ideal solution.


## How?
Strikes out the changes made in #44771 as is, removing anchor support for a total of 53 dynamic blocks. After #44771 was merged, a new Time To Read block was added. However, this block does not currently support anchor, so it does not need to be considered in this PR.

## Testing Instructions

- To test the impact of this PR, the following four types of blocks should be tested:
  - Static block without anchor
  - Static block with anchor
  - Dynamic block without anchor
  - Dynamic block with anchor
- In the `trunk` branch, insert all types of blocks as follows:
   ```html
   
   <p>Static block without anchor</p>
   

   
   <p id="anchor">Static block with anchor</p>
   

   

   
   ```
- Switch to this PR branch and reload the editor. **There should be no blocks to be broken**.
- Open the Advanced tab of the dynamic block and confirm that the HTML ANCHOR is not present.
- Make changes to the content, save the post, and reload. **There should be no blocks to be broken**.
- Open the code editor and confirm that the saved markup has changed as follows:
   ```html
   
   <p>Static block without anchor</p>
   

   
   <p id="anchor">Static block with anchor</p>
   

   

   
   ```
